### ISSUE TITLE
oidc login general error, no error message, login stuck

### ISSUE THREAD

**Issue by raffis** [gyselroth/balloon-client-desktop#72]
#### Description
**Is**:
```
{"category":"startup","level":"debug","message":"startup window opened","timestamp":"2017-12-19T14:09:55.846Z"}
{"category":"openid-connect","request":{"clientId":"xxx","redirectUri":"http://127.0.0.1:13006","scope":"openid profile offline_access","responseType":"code","extras":{"prompt":"consent","access_type":"offline"},"state":"MqL6chbugC"},"response":{"code":"xxx","state":"MqL6chbugC"},"error":null,"level":"info","message":"Authorization request complete ","timestamp":"2017-12-19T14:10:04.715Z"}
{"category":"openid-connect","level":"info","message":"retrieved oauth2 refresh token","timestamp":"2017-12-19T14:10:04.912Z"}
{"category":"openid-connect","error":{"stack":"Error: An unknown error occurred.","message":"An unknown error occurred."},"level":"error","message":"failed to retrieve refreshToken","timestamp":"2017-12-19T14:10:04.961Z"}
{"category":"startup","level":"debug","message":"startup window opened","timestamp":"2017-12-19T14:10:06.679Z"}
{"category":"startup","level":"debug","message":"startup window opened","timestamp":"2017-12-19T14:10:11.863Z"}
```

**Should**:
Error message in auth window.

#### balloon desktop version
0.0.38

#### Your client operating system
OSX

**Comment by raffis** [comment]
probably related to keytar on osx. 
> node_modules//keytar/src/keytar_mac.cc:    errorStr = std::string("An unknown error occurred.");

Maybe if keychain locked?

**Comment by raffis** [comment]
probably also the original error of #69

**Comment by raffis** [comment]
> Maybe if keychain locked?

Probably since re-login resolves the issue.

**Comment by raffis** [comment]
Maybe solved by PR #128

**Comment by pixtron** [comment]
The error "An unknown error occurred." seems to be thrown from keytar. Probably from `clientConfig.storeSecret`

Eventually it would make sense to retry storing the secret, if keytar fails.

**Comment by raffis** [comment]
Looks like this has been fixed with the retry.

**Comment by pixtron** [comment]
Just as a reference, keytar retries implemented with bb73422
